### PR TITLE
Process data

### PR DESCRIPTION
**Pull request by Haywayaheadshot**
# In this milestone I;
> Created models for comment, like, post, and user tables.
> Added an application_record to be able to use ActiveRecord
> Added custom methods to each model.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+  validates :title, length: { maximum: 250 }
+  validates :comments_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
+  validates :likes_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
+
+  def update_posts_counter
+    author.increment(:posts_counter)
+  end
+
```
Comment (julie-ify) [inline app/models/post.rb:15]: - [x] **[OPTIONAL]** We can add the active record callback for this method and make this method private.

  ```
  after_save :update_posts_counter

  private

  def update_posts_counter
    author.increment(:posts_counter)
  end
  ```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, foreign_key: 'author_id', class_name: 'User'
+  belongs_to :post, foreign_key: 'post_id', class_name: 'Post'
+
+  def update_likes_counter
+    post.increment(:likes_counter)
+  end
```
Comment (julie-ify) [inline app/models/like.rb:10]: - [x] **[OPTIONAL]** We can add an active record callback for this method

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, foreign_key: 'author_id', class_name: 'User'
+  belongs_to :post, foreign_key: 'post_id', class_name: 'Post'
+
+  def update_comments_counter
+    post.increment(:comments_counter)
+  end
```
Comment (julie-ify) [inline app/models/comment.rb:10]: - [x] **[OPTIONAL]** We can add an active record callback for this method

**Review by julie-ify — APPROVED** [review]
Hi @Haywayaheadshot,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Haywayaheadshot** [comment]
@julie-ify Thank you for reviewing my project. I will implement the changes before merging.

**Pull request merged**